quiche/src/stream/send_buf.rs

```diff
@@ -0,0 +1,343 @@
+use std::collections::VecDeque;
+
+use crate::ranges::RangeSet;
+
+use crate::Error;
+use crate::Result;
+
+/// Send-side stream buffer.
+///
+/// Stream data scheduled to be sent to the peer and is buffered in a circular
+/// buffer.
+#[derive(Debug, Default)]
+pub struct SendBuf {
+    /// A circular buffer that stores the data that should be sent to the peer
+    pub(super) data: VecDeque<u8>,
```

Comment:
I'm trying to understand the implications here. If I'm understanding correctly, previously, an app write would result in a RangeBuf. Now, we write directly to the VecDeque which will alloc when growth is necessary, but otherwise avoids the allocation. The allocation _does_ eventually happen when the RangeBuf is created to track what needs to be retransmitted. See my previous comment about how this might be avoided. As-is, are we just moving the alloc/copy to a later point in time?

Reply:
I'm now seeing #1399 which I think probably makes more sense